mesonbuild/coredata.py

```diff
@@ -749,26 +756,52 @@ def set_default_options(self, default_options, subproject, env):
             mlog.warning('Recommend using either -Dbuildtype or -Doptimization + -Ddebug. '
                          'Using both is redundant since they override each other. '
                          'See: https://mesonbuild.com/Builtin-options.html#build-type-options')
+
         cmd_line_options = OrderedDict()
-        # Set project default_options as if they were passed to the cmdline.
+        # Set default options as if they were passed to the command line.
         # Subprojects can only define default for user options and not yielding
         # builtin option.
         from . import optinterpreter
-        for k, v in default_options.items():
+        for k, v in chain(default_options.items(), env.meson_options.host.get(subproject, {}).items()):
             if subproject:
                 if (k not in builtin_options or builtin_options[k].yielding) \
                         and optinterpreter.is_invalid_name(k, log=False):
                     continue
-                k = subproject + ':' + k
+                k = '{}:{}'.format(subproject, k)
             cmd_line_options[k] = v
 
-        # Override project default_options using conf files (cross or native)
-        for k, v in env.paths.host:
-            if v is not None:
```

Comment:
[paths] is deprecated but should still work, are you sure that code can be deleted?

Reply:
nevermind, you load them into user_options, good.